A workaround for error when building for riscv64.

Upstream report: https://bugreports.qt.io/browse/QTBUG-84580

Patch: https://bugs.gentoo.org/803470


--- a/qtbase/mkspecs/common/linux.conf
+++ b/qtbase/mkspecs/common/linux.conf
@@ -33,7 +33,7 @@ QMAKE_LIBS_EGL          = -lEGL
 QMAKE_LIBS_OPENGL       = -lGL
 QMAKE_LIBS_OPENGL_ES2   = -lGLESv2
 QMAKE_LIBS_OPENVG       = -lOpenVG
-QMAKE_LIBS_THREAD       = -lpthread
+QMAKE_LIBS_THREAD       = -pthread
 QMAKE_LIBS_VULKAN       =
 
 QMAKE_INCDIR_WAYLAND    =

--- a/qtbase/src/corelib/configure.json
+++ b/qtbase/src/corelib/configure.json
@@ -355,7 +355,7 @@
                     "std::future<int> f = std::async([]() { return 42; });",
                     "(void)f.get();"
                 ],
-                "qmake": "unix:LIBS += -lpthread"
+                "qmake": "unix:LIBS += -pthread"
             }
         },
         "cxx11_random": {
@@ -466,7 +466,7 @@
                     "shm_open(\"test\", O_RDWR | O_CREAT | O_EXCL, 0666);",
                     "shm_unlink(\"test\");"
                 ],
-                "qmake": "linux: LIBS += -lpthread -lrt"
+                "qmake": "linux: LIBS += -pthread -lrt"
             }
         },
         "linkat": {
